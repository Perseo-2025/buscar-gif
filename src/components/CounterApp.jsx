import PropTypes from "prop-types"
import { useState } from "react"

export default function CounterApp({title, subtitle, value}) {

    const [counter, setCounter] = useState(value)

    const handleAdd = () => setCounter(counter + 1)
    const handleReset = () => setCounter(value )
    
    const handleSub = () => setCounter(counter - 1)

  return (
    <div>

        <h1 className="text-3xl font-bold" data-testid="test-title">{title}</h1>
        <p>{subtitle}</p>
        <h2 className="text-2xl font-bold p-2 m-1 bg-indigo-300 "> {counter} </h2>
        <button className="p-1 m-1 bg-amber-200 rounded hover:cursor-pointer" onClick={handleAdd}> +1 </button>
        <button className="p-1 m-1 bg-amber-200 rounded hover:cursor-pointer" onClick={handleSub}> -1 </button>
        <button aria-label="btn-reset"  className="p-1 m-1 bg-amber-200 rounded hover:cursor-pointer" onClick={handleReset}> Reset </button>
    </div>
  )
}

CounterApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired
}
CounterApp.defaultProps = {
  value: 0,
}