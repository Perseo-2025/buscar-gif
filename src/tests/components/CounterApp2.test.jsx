import { fireEvent, render, screen } from "@testing-library/react"
import CounterApp from "../../components/CounterApp"

describe('Pruebas en <CounterApp />', () => {
    
    const title = 'Rey Perseo';
    const subtitle = 'Cardenas Sanchez'

    test('debe de hacer match con el snapshot', () => {
        const {container} = render(<CounterApp title={title} />)
        expect(container).toMatchSnapshot()
    })
    test('debe de mostrar el mensaje Rey perseo', () => {
        render(<CounterApp title={title}/>)
        expect(screen.getByText(title)).toBeTruthy()
    })
    test('debe de mostar el título en un h1', () => {
        render(<CounterApp title={title}/>)
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)
    })
    test('debe de mostrar el subtitulo enviado por props', () => {
        render(<CounterApp title={title} subtitle={subtitle}/>)
        expect(screen.getAllByText(subtitle).length).toBe(1)
    })

    test('debe de incrementar con el botón +1', () => {
        const initialValue = 100
        render(<CounterApp value={initialValue}/>)
        fireEvent.click( screen.getByText('+1') )
        expect(screen.getByText('101')).toBeTruthy()
    })
    test('debe de decrementar con el botón -1', () => {
        const initialValue = 100
        render(<CounterApp value={initialValue}/>)
        fireEvent.click( screen.getByText('-1') )
        expect(screen.getByText('99')).toBeTruthy()
    })
    
    test('debe de funcionar el botón de reset', () => {
        const initialValue = 100
        render(<CounterApp value={initialValue}/>)
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+Reset') )

        fireEvent.click(('button', {name: 'btn-reset'} ))
        
        expect(screen.getByText(initialValue)).toBeTruthy()

    })
})