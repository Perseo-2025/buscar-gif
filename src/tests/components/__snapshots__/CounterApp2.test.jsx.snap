// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1
      class="text-3xl font-bold"
      data-testid="test-title"
    >
      Rey Perseo
    </h1>
    <p />
    <h2
      class="text-2xl font-bold p-2 m-1 bg-indigo-300 "
    >
       
       
    </h2>
    <button
      class="p-1 m-1 bg-amber-200 rounded hover:cursor-pointer"
    >
      +1
    </button>
    <button
      class="p-1 m-1 bg-amber-200 rounded hover:cursor-pointer"
    >
      Reset
    </button>
    <button
      class="p-1 m-1 bg-amber-200 rounded hover:cursor-pointer"
    >
      -1
    </button>
  </div>
</div>
`;
